theano/tensor/var.py

```diff
@@ -533,7 +534,12 @@ def includes_bool(args_el):
         axis = None
         for i, arg in enumerate(args):
             try:
-                if arg is not np.newaxis:
+                if (isinstance(arg, (np.ndarray, theano.tensor.Variable)) and
```

Comment:
I moved the test to Subtensor.convert() now.

It still requires a test here, though: if there is a boolean mask, it should be given to AdvancedSubtensor and not AdvancedSubtensor1, even if there is only one mask.